core/src/main/java/com/sequenceiq/cloudbreak/service/SimpleSecurityService.java

```diff
@@ -0,0 +1,215 @@
+package com.sequenceiq.cloudbreak.service;
+
+import static com.sequenceiq.cloudbreak.EnvironmentVariableConfig.CB_CERT_DIR;
+import static com.sequenceiq.cloudbreak.EnvironmentVariableConfig.CB_TLS_CERT_FILE;
+import static com.sequenceiq.cloudbreak.EnvironmentVariableConfig.CB_TLS_PRIVATE_KEY_FILE;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+import javax.inject.Inject;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.KeyPair;
+import com.sequenceiq.cloudbreak.core.CloudbreakSecuritySetupException;
+import com.sequenceiq.cloudbreak.domain.Stack;
+import com.sequenceiq.cloudbreak.orchestrator.GatewayConfig;
+import com.sequenceiq.cloudbreak.repository.StackRepository;
+import com.sequenceiq.cloudbreak.service.stack.flow.TLSClientConfig;
+import com.sequenceiq.cloudbreak.util.FileReaderUtils;
+
+@Component
+public class SimpleSecurityService {
```

Comment:
I'll rename the service to something other than Simple. Regarding the temp files - the third party docker library that we use needs to have the keys in a directory (even with specific naming conventions) to be able to use them.